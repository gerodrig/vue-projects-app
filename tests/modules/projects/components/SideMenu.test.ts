import { nextTick } from 'vue';
import { createTestingPinia } from '@pinia/testing';

import { mount } from '@vue/test-utils';
import SideMenu from '@/modules/projects/components/SideMenu.vue';
import { useProjectsStore } from '@/modules/projects/store/projects.store';
import { fakeProjects } from '../../../mocks/projects.fake';

describe('<SideMenu />', () => {
  const wrapper = mount(SideMenu, {
    global: {
      stubs: ['RouterLink'],
      plugins: [createTestingPinia()],
    },
  });

  const store = useProjectsStore();

  beforeEach(() => {
    store.$patch({
      projects: [],
    });
  });

  it('Should render with no projects', () => {
    expect(wrapper.html()).toContain('There are no projects available');
  });

  it('Should render with projects', async () => {
    store.$patch({
      projects: fakeProjects,
    });

    await nextTick();

    expect(wrapper.html()).not.toContain('There are no projects available');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
