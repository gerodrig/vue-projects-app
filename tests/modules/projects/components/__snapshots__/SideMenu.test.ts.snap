// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SideMenu /> > Should render with projects 1`] = `
"<aside class="bg-base-200 w-72 min-h-screen">
  <h2 class="text-lg font-bold mx-4">Projects</h2><!-- Menu -->
  <ul class="menu">
    <li>
      <details>
        <summary>
          <router-link-stub to="/project/1"></router-link-stub>
        </summary>
        <ul>
          <li>
            <router-link-stub to="/project/1"></router-link-stub>
          </li>
          <li>
            <router-link-stub to="/project/1"></router-link-stub>
          </li>
          <li>
            <router-link-stub to="/project/1"></router-link-stub>
          </li>
          <li>
            <router-link-stub to="/project/1"></router-link-stub>
          </li>
        </ul>
      </details>
    </li>
    <li>
      <router-link-stub to="/project/2"></router-link-stub>
    </li>
    <li>
      <details>
        <summary>
          <router-link-stub to="/project/3"></router-link-stub>
        </summary>
        <ul>
          <li>
            <router-link-stub to="/project/3"></router-link-stub>
          </li>
          <li>
            <router-link-stub to="/project/3"></router-link-stub>
          </li>
        </ul>
      </details>
    </li>
    <li>
      <details>
        <summary>
          <router-link-stub to="/project/4"></router-link-stub>
        </summary>
        <ul>
          <li>
            <router-link-stub to="/project/4"></router-link-stub>
          </li>
          <li>
            <router-link-stub to="/project/4"></router-link-stub>
          </li>
          <li>
            <router-link-stub to="/project/4"></router-link-stub>
          </li>
        </ul>
      </details>
    </li>
  </ul>
</aside>"
`;
